Code/Framework/AzCore/AzCore/Serialization/SerializationUtils.cpp

```diff
@@ -527,6 +527,60 @@ namespace AZ::Utils
         return ptr;
     }
 
+    bool InspectSerializedFile(
+        const char* filePath,
+        SerializeContext* sc,
+        const ObjectStream::ClassReadyCB& classCallback,
+        Data::AssetFilterCB assetFilterCallback)
+    {
+        if (!AZ::IO::FileIOBase::GetInstance()->Exists(filePath))
+        {
+            AZ_Error("Verify", false, "Unable to open file '%s' as it doesn't exist.", filePath);
+            return false;
+        }
+
+        AZ::IO::HandleType fileHandle;
+        auto openResult = AZ::IO::FileIOBase::GetInstance()->Open(filePath, AZ::IO::OpenMode::ModeRead, fileHandle);
+        if (!openResult)
+        {
+            AZ_Error("Verify", false, "File '%s' could not be opened.", filePath);
+            return false;
+        }
+
+        u64 fileLength = 0;
+        auto sizeResult = AZ::IO::FileIOBase::GetInstance()->Size(fileHandle, fileLength);
+        if (!sizeResult || (fileLength == 0))
+        {
+            AZ_Error("Verify", false, "File '%s' doesn't have content.", filePath);
+            return false;
+        }
+
+        AZStd::vector<u8> data;
+        data.resize_no_construct(fileLength);
+        u64 bytesRead = 0;
+        auto readResult = AZ::IO::FileIOBase::GetInstance()->Read(fileHandle, data.data(), fileLength, true, &bytesRead);
+        if (!readResult || (bytesRead != fileLength))
+        {
+            AZ_Error("Verify", false, "Unable to read file '%s'.", filePath);
+            return false;
+        }
+
+        AZ::IO::FileIOBase::GetInstance()->Close(fileHandle);
+
+        AZ::IO::MemoryStream stream(data.data(), fileLength);
```

Comment:
As the ObjectStream accepts an `AZ::IO::GenericStream` derived class, this logic can be simplified to just opening up an AZ::IO::FileIOStream.

```suggestion
        AZ::IO::FileIOStream stream(filePath, AZ::IO::OpenMode::ModeRead);
        if (!stream.IsOpen())
        {
            AZ_Error("Verify", false, "File '%s' could not be opened.", filePath);
            return false;
        }
```